### ISSUE TITLE
missing-assertion doesn't raise warning when return+expect without assertion

### ISSUE THREAD

**Issue by diego-G** [Turbo87/eslint-plugin-chai-expect#37]
`missing-assertion` rule doesn't raise any warning when there is a `return` combined with an `expect` without assertion.

```node
	describe('calculateFee', () => {
		it('should return the correct fee for delegate transaction', () => {
                         // Missing assertion false positive
			return expect(delegate.calculateFee(transaction).isEqualTo(FEES.DELEGATE));
		});
	});
```

**Comment by Turbo87** [comment]
why do you use `return` there? I don't understand that code snippet

**Comment by diego-G** [comment]
The correct snippet is as follows:
```node
describe('calculateFee', () => {
	it('should return the correct fee for delegate transaction', () => {
		return expect(delegate.calculateFee(transaction).isEqualTo(FEES.DELEGATE)).to.be.true;
	});
});
```

**Comment by diego-G** [comment]
We use https://github.com/lo1tuma/eslint-plugin-mocha/tree/4b7963346efed34a46dc51fa2ad7958b8f3cb2e4/docs/rules and by default this rule [no-synchronous-tests](https://github.com/lo1tuma/eslint-plugin-mocha/blob/4b7963346efed34a46dc51fa2ad7958b8f3cb2e4/docs/rules/no-synchronous-tests.md).

**Comment by Turbo87** [comment]
I still don't understand. why do you return anything from that function?

**Comment by diego-G** [comment]
Let's pick another example:
https://github.com/LiskHQ/lisk/blob/7eb913eb8f89f9bbd9107f25905e8d4cc4a4f69b/test/unit/logic/account.js#L572
If we remove `.to.equal(50)` from that line, your rule overlooks the missed assertion case.

**Comment by Turbo87** [comment]
@diego-G maybe, but in that line the `return` itself doesn't make any sense

**Comment by diego-G** [comment]
It is the way we write tests and it is supported by Mocha and Chai. I believe it is a fair possible case for your rule. I've linked this issue to ours.

**Comment by Turbo87** [comment]
I personally don't see the use case and won't spend any time on it, but feel free to work on it and open a PR
